discord_bot2.py

```diff
@@ -163,8 +164,105 @@ async def on_ready(self):
         logging.info(self.user.name)
         logging.info(self.user.id)
         logging.info('------')
+        channels = self.get_all_channels()
+        for channel in channels:
+            if channel.name == 'control-duck':
+                await channel.send('Duck online')
 
-    async def on_message(self, message):
+    async def restart(self, message):
+        """
+        Restart the bot
+        :param message: The message that triggered the restart
+        """
+        await message.channel.send(f'Restart requested.')
+        os.chdir(Path(__file__).parent)
+        await self.execute_command('git fetch', message.channel)
+        await self.execute_command('git reset --hard', message.channel)
+        await self.execute_command('git clean -f', message.channel)
+        await self.execute_command('git pull --rebase=false', message.channel)
+        await self.execute_command('poetry install', message.channel)
+        await message.channel.send(f'Restarting.')
+        subprocess.Popen(["bash", "restart.sh"])
+        return
+
+    async def say_in_channel(self, channel, text, prefix="", suffix=""):
+        """
+        Say something in a channel
+        :param channel: The channel to say something in
+        :param output: The message to say, long messages will be split into multiple messages
+        """
+        # Async methods like for loops better than while loops
+        # Split output into 1800 character chunks
+        for i in range(len(text) // 1800):
+            await channel.send(f'{prefix}{text[:1800]}{suffix}')
+            text = text[1800:]
+
+        await channel.send(f'{prefix}{text}{suffix}')
+
+    async def execute_command(self, text, channel):
+        """
+        Execute a command in the shell and return the output to the channel
+        """
+        # Run command using shell and pipe output to channel
+        await self.say_in_channel(channel, text, "```ps\n$ ", "```")
+        process = subprocess.run(text, shell=True, stdout=subprocess.PIPE, stderr=subprocess.PIPE)
+        # Get output of command and send to channel
+        errors = process.stderr.decode('utf-8')
+        if errors:
+            await self.say_in_channel(channel, errors, prefix=f'Errors: ```', suffix='```')
+        output = str(process.stdout.decode('utf-8'))
+        if output:
+            await self.say_in_channel(channel, output, prefix=f'```', suffix='```')
+        if len(output) > 2000:
+            await channel.send(f'Done.')
+        return
+
+    async def execute_message(self, message):
+        """
+        Execute a command in the shell and return the output to the channel
+        Removes the starting !
+        :param message: discord.Message object, used to get the command
+        """
+        await self.execute_command(message.content[1:], message.channel)
+        return
```

Comment:
This method seems like a security nightmare waiting to happen. It doesn't look like it is used though (which is good!). Is it left over from before?